test/spec_files.rb

```diff
@@ -73,6 +73,15 @@ def files(*args)
     res["last-modified"].must_equal File.mtime(path).httpdate
   end
 
+  it "supports case-insensitive header access" do
+    res = Rack::MockRequest.new(files(DOCROOT)).get("/cgi/test")
+
+    path = File.join(DOCROOT, "/cgi/test")
+
+    res.must_be :ok?
+    res["Last-Modified"].must_equal File.mtime(path).httpdate
```

Comment:
Consider adding a comment here explaining that you are explicitly testing that the non-normalised upper case form of then header is being used.